spec/scheduling-tasks.md

```diff
@@ -272,49 +362,105 @@ Issue: [=Run steps after a timeout=] doesn't necessarily account for suspension;
 [whatwg/html#5925](https://github.com/whatwg/html/issues/5925).
 
 <div algorithm>
-  To <dfn>select the scheduler task queue</dfn> for a {{Scheduler}} |scheduler| given an
-  {{AbortSignal}} or null |signal|, and a {{TaskPriority}} or null |priority|:
-
-  1. If |priority| is null, |signal| is not null and [=implements=] the {{TaskSignal}} interface,
-     and |signal| [=TaskSignal/has fixed priority=], then set |priority| to |signal|'s
-     [=TaskSignal/priority=].
-  1. If |priority| is null and |signal| is not null and [=implements=] the {{TaskSignal}} interface,
-     then
+  To <dfn>schedule a yield continuation</dfn> for {{Scheduler}} |scheduler| given
+  {{SchedulerYieldOptions}} |options|:
+
+  1. Let |result| be [=a new promise=].
+  1. Let |abortOption| be |options|["{{SchedulerYieldOptions/signal}}"] if it [=map/exists=],
+     otherwise null.
+  1. Let |priorityOption| be |options|["{{SchedulerPostTaskOptions/priority}}"] if it
+     [=map/exists=], otherwise null.
+  1. If |abortOption| is null and |priorityOption| is null, then set |abortOption| to
+     "{{SchedulerSignalInherit/inherit}}" and set |priorityOption| to
+     "{{SchedulerSignalInherit/inherit}}".
+  1. Otherwise if |abortOption| is "{{SchedulerSignalInherit/inherit}}" and |priorityOption| is null,
+     then set |priorityOption| to "{{SchedulerSignalInherit/inherit}}".
+  1. Let |state| be the result of [=computing the scheduling state from options=] given |scheduler|,
+     |abortOption|, and |priorityOption|.
+  1. Let |signal| be |state|'s [=scheduling state/abort source=].
+  1. If |signal| is not null and it is [=AbortSignal/aborted=], then [=reject=] |result| with
+     |signal|'s [=AbortSignal/abort reason=] and return |result|.
+  1. Let |handle| be the result of [=creating a task handle=] given |result| and |signal|.
+  1. If |signal| is not null, then [=AbortSignal/add=] |handle|'s [=task handle/abort steps=] to
+     |signal|.
+  1. Set |handle|'s [=task handle/queue=] to the result of [=selecting the scheduler task queue=]
+     for |scheduler| given |state|'s [=scheduling state/priority source=] and true.
+  1. [=Schedule a task to invoke an algorithm=] for |scheduler| given |handle| and the following
+     steps:
+    1. Resolve |result|.
+  1. Return |result|.
+</div>
+
+<div algorithm>
+  To <dfn>compute the scheduling state from options</dfn> given a {{Scheduler}} object |scheduler|,
```

Comment:
Overall, I find this algorithm hard to follow; I've written and deleted at least three comments where I thought there were errors already :)

Something structured like this might be clearer:

1. Let result be...
2. Let inheritedState be...
3. If abortOption is "inherit":
    * if inheritedState is not null, then set result’s abort source to inheritedState’s abort source.
4. Otherwise, set result’s abort source to abortOption.
5. If priorityOption is "inherit":
    * if inheritedState is not null, then set result’s priority source to inheritedState’s priority source.
6. Otherwise, if priorityOption is not null, then set result’s priority source to the result of creating a fixed priority unabortable task signal given priorityOption.
7. Otherwise, if abortOption is not null and implements the TaskSignal interface, then set result’s priority source to abortOption. _(although I would have expected this to reference result's abort source, rather than abortOption here)_
8. If result’s priority source is null, then set result’s priority source to the result of creating a fixed priority unabortable task signal given "user-visible".
9. Return result.

Reply:
Heh, I want back and forth on this several times when drafting this. IIUC this was mostly rewriting 3 & 5? Just making sure I didn't miss anything (changed those in latest version).

Re: using `abortOption` in step 7 -- this handles the case where the arguments are of the form `{signal: someSignal}`, where there's no inheritance and the signal could be either a TaskSignal or plain AbortSignal. If it's a TaskSignal, then that priority gets used; otherwise, it defaults to "user-visible" in step 8.

Reply:
Thanks -- I missed that possibility